### PR TITLE
Module 3

### PR DESCRIPTION
**Pull request by jatin-28**

### CODE REVIEW

File: tests/test_selenium_app.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    driver.get('http://localhost:5000/')
+    assert driver.title == 'Quick and Dirty To-Do'
+
+    # Create an item
+    todo_item = 'Testing a todo item'
+    inputElement = driver.find_element_by_id("newitem")
+    inputElement.send_keys(todo_item)
+    inputElement.submit()
+
+    driver.implicitly_wait(2)  # seconds
```
Comment (hugh-emerson) [inline tests/test_selenium_app.py:110]: You don't need to keep calling `implicitly_wait`
You do it once to set up the driver and all future interactions will implicitly wait up that many seconds for the element to appear

File: tests/test_app.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# def test_add_item_with_a_todo(mock_post_request, client):
+#     todos_json = {
+#             "id": "5f57f133fe6fec1ff7e25d63",
+#             "dateLastActivity": "2020-09-21T17:39:25.603Z",
+#             "name": "Valid Item",
+#             "due": None
+#         }
+#
+#     response = client.post('/additem')
+#     mock_post_request.assert_called_with('/cards')
```
Comment (hugh-emerson) [inline tests/test_app.py]: It's fine if you gave up on this test, but you can delete the commented out code now or let me know if you want help getting it to work

File: common.py [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
 class TodoItem:
-    def __init__(self, title, status, id=None, duedate=None):
+    def __init__(self, title, status, id=None, duedate=None, last_modified=datetime.now().isoformat()):
+        self.last_modified = dateutil.parser.parse(last_modified)
         self.id = id
         self.title = title
         self.status = status
-        self.duedate = duedate
\ No newline at end of file
+        self.duedate = duedate
```
Comment (hugh-emerson) [inline common.py:48]: Your types are inconsistent - self.duedate is a string, but self.last_modified is a datetime, right?

Datetime for both would be better

File: app.py [hunk 4]
```diff
@@ -3,60 +3,73 @@
...
 
-# Jinja filters
-@app.template_filter()
-def format_datetime(value):
-    fromisoformat = dateutil.parser.parse(value)
-    return fromisoformat.strftime("%b %d")
 
+def create_app():
+    app = Flask(__name__, static_url_path='/static')
+    app.config.from_object('flask_config.Config')
```
Comment (hugh-emerson) [inline app.py:16]: Unrelated to this PR's changes, but your app doesn't actually use TODO_BOARD_ID - you set it but never use it for any requests. So any reference to it should be removed.

You should also remove TODO_LIST_ID etc from flask_config.py as they're not used from there. You use os.getenv again elsewhere

**Review by hugh-emerson — APPROVED** [review]
Looks good! I've left some suggestions but you can merge whenever you're ready

File: tests/test_app.py (hunk 2 above)
Comment (jatin-28) [reply tests/test_app.py]: Must have forgotten! Done it!

File: app.py (hunk 4 above)
Comment (jatin-28) [reply app.py:16]: Refactored and removed the LIST ID completely and now it uses the TODO_BOARD_ID

File: common.py (hunk 3 above)
Comment (jatin-28) [reply common.py:48]: I'll keep this in mind for the next refactor.

**Pull request merged**